### DISCUSSION TITLE
Academy - What are requirements for preview data source?

### DISCUSSION THREAD

**Discussion by jantudjan99** [pimcore/pimcore#16342] [Q&A / Support]
Hello,
I'm struggling with figuring out and to pass this question. Can somebody help me and tell me why are my answers incorrect. Which ones are correct? 
![image](https://github.com/pimcore/pimcore/assets/58600220/a89030a9-6217-4df9-9fe4-8122db345abc)
Thanks in advance!

**Comment by jtwogw** [comment]
Have you tried 1 & 4? 

2) NOT: It supports any file format with custom adapters
3) NOT: loading all records is not recommended for preview
